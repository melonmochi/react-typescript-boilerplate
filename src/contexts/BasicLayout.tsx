import React, { FC, createContext, useReducer } from "react";
import { BasicLayout as BasicLayoutType } from "typings";
import { basicLayoutReducers as reducers } from "@/reducers";

const initialState: BasicLayoutType.State = {
  collapsed: false,
  settings: {
    colorWeak: false,
    contentWidth: "Fluid",
    fixSiderbar: false,
    fixedHeader: false,
    layout: "sidemenu",
    menu: { locale: true },
    navTheme: "dark",
    title: "React TypeScript"
  }
};

export const Context = createContext<BasicLayoutType.Interface>({
  state: initialState,
  dispatch: () => {}
});

export const ContextProvider: FC = props => {
  const [state, dispatch] = useReducer(reducers, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>
      {props.children}
    </Context.Provider>
  );
};
