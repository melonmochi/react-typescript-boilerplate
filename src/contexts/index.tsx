export {
  Context as BasicLayoutContext,
  ContextProvider as BasicLayoutContextProvider
} from "./BasicLayout";
