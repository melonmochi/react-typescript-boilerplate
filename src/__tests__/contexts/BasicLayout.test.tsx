import * as React from "react";
import { shallow } from "enzyme";
import { BasicLayoutContextProvider } from "@/contexts";

describe("BasicLayoutContextProvider", () => {
  test("Snapshot", () => {
    const basicLayout = shallow(<BasicLayoutContextProvider />);
    expect(basicLayout).toMatchSnapshot();
  });
});
