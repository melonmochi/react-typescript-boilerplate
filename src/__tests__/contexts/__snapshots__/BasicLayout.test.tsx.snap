// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLayoutContextProvider Snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "globalRef": Object {
        "current": null,
      },
      "state": Object {
        "collapsed": false,
        "settings": Object {
          "colorWeak": false,
          "contentWidth": "Fluid",
          "fixSiderbar": false,
          "fixedHeader": false,
          "layout": "sidemenu",
          "menu": Object {
            "locale": true,
          },
          "navTheme": "dark",
          "title": "React TypeScript",
        },
      },
    }
  }
/>
`;
